packages/runtime/container-runtime/src/connectionTelemetry.ts

```diff
@@ -18,21 +18,43 @@ import { assert, performance } from "@fluidframework/common-utils";
  */
 export const latencyThreshold = 5000;
 
+interface IOpPerfTelemetryProperties {
+    /** Track how long op is sitting in inbound queue until it is processed */
+    durationInboundQueue: number | undefined;
+    /** Measure time outbound op is sitting in queue due to active batch */
+    durationOutboundQueue: number | undefined;
+    /** Time spent between DeltaManager's inbound "push" event until the DeltaManager "op" event */
+    durationInboundToProcessing: number | undefined;
+    /** Length of the DeltaManager's inbound queue at the time of the processing */
+    lenghtInboundQueue: number | undefined;
+}
+/**
+ * We collect the timing at various moments in time during the op processing.
+ */
+enum OpTimings {
+    opSendTimeForLatencyStatistics,
+    opTimeSittingInboundQueue,
+    opTimeInboundPushEvent,
+}
 class OpPerfTelemetry {
     private pongCount: number = 0;
     private pingLatency: number | undefined;
 
     // Collab window tracking. This is timestamp of %1000 message.
     private sequenceNumberForMsnTracking: number | undefined;
     private msnTrackingTimestamp: number = 0;
-
-    // To track round trip time for every %1000 client message.
-    private opSendTimeForLatencyStatistics: number | undefined;
-
+    // To track round trip time for every %500 client message.
     private clientSequenceNumberForLatencyStatistics: number | undefined;
 
-    private opTimeSittingInInboundQueue: number | undefined;
-    private durationSittingInInboundQueue: number | undefined;
+    private readonly opProcessingTimes: number | undefined[] = [];
```

Comment:
I'm not sure why you need an array & OpTimings. I think an object (i.e. with property names and values) is more natural way to do it and type safe.

Reply:
Also, these are not times (aka duration), these are timestamps for start of events. I'd call it opStartTimes.

Reply:
changed to startTimes.

Reply:
As for the Object, Is it ok to create as Partial<Interface> ?